src/main/scala/pirate/Flags.scala

```diff
@@ -24,11 +24,11 @@ trait Flags {
     parse(SwitchParser(meta, true)) ||| false.pure[Parse]
 
   def flag[A: Read](meta: Metadata): Parse[A] =
```

Comment:
I think this will need some more negative test cases to validate. The only real difference is where the "Fail" part ends-up. We should be validating these cases and at least making sure they don't change or we understand the change. I can't say I do, just looking at the code.